client/client.go

```diff
@@ -366,33 +357,50 @@ func dialOpenSSL(conf Config, network string, addr string) (net.Conn, error) {
 	if v != openssl.Ok {
 		if v == openssl.CertHasExpired {
 			return nil, errors.Errorf("certificate has expired, "+
-				"openssl verify rc: %d server cert file: %s", v, conf.ServerCert)
+				"openssl verify rc: %d server cert file: %s", v, serverCert)
 		}
 		if v == openssl.DepthZeroSelfSignedCert {
 			return nil, errors.Errorf("depth zero self-signed certificate, "+
-				"openssl verify rc: %d server cert file: %s", v, conf.ServerCert)
+				"openssl verify rc: %d server cert file: %s", v, serverCert)
 		}
 		if v == 0x42 { //X509_V_ERR_EE_KEY_TOO_SMALL
 			return nil, errors.Errorf("end entity key too short, "+
-				"openssl verify rc: %d server cert file: %s", v, conf.ServerCert)
+				"openssl verify rc: %d server cert file: %s", v, serverCert)
 		}
 		if v == 0x14 { //X509_V_ERR_UNABLE_TO_GET_ISSUER_CERT_LOCALLY
 			return nil, errors.Errorf("certificate signed by unknown authority, "+
-				"openssl verify rc: %d server cert file: %s", v, conf.ServerCert)
+				"openssl verify rc: %d server cert file: %s", v, serverCert)
 		}
 		return nil, errors.Errorf("not a valid certificate, "+
-			"openssl verify rc: %d server cert file: %s", v, conf.ServerCert)
+			"openssl verify rc: %d server cert file: %s", v, serverCert)
 	}
 	return conn, err
 }
 
 func newHttpsClient(conf Config) (*http.Client, error) {
 	client := newHttpClient()
 
+	ctx, err := openssl.NewCtx()
+	if err != nil {
+		return nil, err
+	}
+
+	ctx, err = loadServerTrust(ctx, &conf)
```

Comment:
Is this needed inside `newHttpsClient`? It seems improper in here.

Reply:
Never mind. I misread it, I thought `newHttpsClient` was referring to "client certificate", but it's of course talking about Mender, the client.